src/App.tsx

```diff
@@ -0,0 +1,152 @@
+    const [dropdawnOptions, setDropdawnOptions] = useState<{ value: Units, label: string }[]>(
+        [
+            { value: 'metric', label: 'Metric, °C' },
+            { value: 'imperial', label: 'Imperial, °F' },
+            { value: 'standard', label: 'Standard, K' }
+        ])
+
+    const isOffline = true
+
+    useEffect(() => {
+        if (isOffline) {
+            return
+        }
+        setLoadStatus(LOAD_STATUSES.LOADING)
+        myFetch(`https://api.openweathermap.org/data/2.5/weather?q=${searchCity}&appid=${process.env.REACT_APP_OPEN_WEATHER_TOKEN}&units=${unit}`)
+
+            .then((weather) => setWeather(weather))
+            .then(() => { setLoadStatus(LOAD_STATUSES.LOADED) })
+            .catch(() => { setLoadStatus(LOAD_STATUSES.ERROR) })
+    }, [])
+
+    useEffect(() => {
+        if (isOffline) {
+            return
+        }
+        setLoadStatus(LOAD_STATUSES.LOADING)
+        myFetch(`https://api.openweathermap.org/data/2.5/weather?q=${searchCity}&appid=a7e03ffabe5b1e62a91464877799652d&units=${unit}`)
+            .then((weather) => setWeather(weather))
+            .then(() => { setLoadStatus(LOAD_STATUSES.LOADED) })
+            .catch(() => { setLoadStatus(LOAD_STATUSES.ERROR) })
+    }, [searchCity, unit])
+
+
+    const infoItems: { icon?: any; label: string; key: string; unit: UnitsLabel }[] = [
+        {
+            icon: humIcon,
+            label: "Humidity",
+            key: 'humidity',
+            unit: {
+                metric: ' %',
+                imperial: " %",
+                standard: " %"
+            }
+        },
+        {
+            icon: pressureIcon,
+            label: "Pressure",
+            key: 'pressure',
+            unit: {
+                metric: ' gPa',
+                imperial: " gPa",
+                standard: " gPa"
+            }
+        },
+        {
+            icon: windIcon,
+            label: "Wind",
+            key: 'speed',
+            unit: {
+                metric: ' m/s',
+                imperial: " miles/hour",
+                standard: " m/s"
+            }
+        },
+    ];
+    const unitLabels = {
+        metric: "°C",
+        imperial: "°F",
+        standard: "K",
+    }
+
+    return (
+        <div className={css.main}>
+            <Loader isLoading={loadStatus === LOAD_STATUSES.LOADING} />
+            {loadStatus === LOAD_STATUSES.ERROR && (<span>Что-то пошло не так...</span>)}
+            <div className={css.container_left}>
+                <div className={css.logo}>
+                    <img className={css.weatherIcon} src={`https://openweathermap.org/img/wn/${weather.weather[0].icon}@2x.png`} alt='weather icon' />
+                </div>
+                <div className={css.infoWeather}>
+                    <p className={css.temperature}>
+                        {/* @ts-ignore */}
+                        {Math.round(weather?.main.temp!)} {unitLabels[unit]}
```

Comment:
тут и ниже ругается на возможный undefined, потому что ты погоду опциональной сделал.
```suggestion
                        {Math.round(weather.main.temp)} {unitLabels[unit]}
```